[MiloTruck] iP

MiloTruck opened the pull request:
# MiluTrock
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

MiluTrock helps you to organize your details and planning. It is:

* text-based
* fast and easy to learn
* ~fast~ _**blazingly fast**_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add in your tasks!!
4. then let it manage your tasks for you!! 😎✨

And it comes at a cheap cost of $9.99!

Here are some features:

* [x]  managing tasks 
* [ ] notifications (coming soon)
* [ ]  managing reminders **(coming soon)**

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
    public static void main(String[] args) {
        new MiluTrock("./data.txt").run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Add greeting

Commit:
Add echo user input

Commit:
Implement ability to store and list input

Commit:
Add mark as done functionality

Commit:
Add override annotation

Commit:
Add task types

Commit:
Replace console with scanner for input

Commit:
Add automated testing

Commit:
Use static factory pattern

Commit:
Add error handling

Commit:
Add delete functionality

Commit:
Bug fix

Commit:
Add task list storage

MiloTruck committed:
Merge pull request #1 from MiloTruck/branch-Level-7

branch-Level-7

Commit:
Add datetime support

MiloTruck committed:
Merge pull request #2 from MiloTruck/branch-Level-8

Add datetime support

Commit:
A-MoreOOP

Commit:
Organize classes into packages

MiloTruck committed:
Merge pull request #3 from MiloTruck/add-gradle-support

Add gradle support

Commit:
Configure gradle

Commit:
Add TaskList and Parser tests

Commit:
Add support for Java 11

Commit:
Add vscode to .gitignore

Commit:
Add JavaDoc header comments

Commit:
Refactor to following coding standard

Commit:
Add find functionality

MiloTruck committed:
Merge pull request #4 from MiloTruck/branch-Level-9

Add find functionality

MiloTruck committed:
Merge branch 'master' into branch-A-CodingStandard

MiloTruck committed:
Merge pull request #5 from MiloTruck/branch-A-CodingStandard

Refactor to following coding standard

MiloTruck committed:
Merge branch 'master' into branch-A-JavaDoc

MiloTruck committed:
Merge pull request #6 from MiloTruck/branch-A-JavaDoc

Add JavaDoc header comments

Commit:
Add JavaDoc header comments for level 9

Commit:
Refactor instance variable references

publicmain commented on src/main/java/milutrock/Parser.java:
```diff
@@ -79,6 +83,12 @@ private void handleDelete(String[] words) {
         ui.printDeleteMessage(task);
     }
 
+    private void handleFind(String input) {
```
The method name `handleFind` is clear and descriptive, indicating that the method handles the find operation. However, I noticed that `handleDelete` and `handleAdd` are also following this pattern. Consistency in using action verbs like "handle" for method names is good practice. It clearly indicates the purpose of these methods. Great job on keeping the naming consistent and descriptive!

publicmain commented on src/main/java/milutrock/Parser.java:
```diff
@@ -79,6 +83,12 @@ private void handleDelete(String[] words) {
         ui.printDeleteMessage(task);
```
The method `printFindMessage` is clear in its intent, but perhaps it could be renamed to something like `displayMatchingTasks` to better reflect its purpose of showing a list of matching tasks. The term "print" might imply outputting to a console, while "display" is a bit more general and can apply to various forms of output.

publicmain commented on src/main/java/milutrock/TaskList.java:
```diff
@@ -19,11 +19,11 @@ public Task removeTask(int i) {
+        this.tasks.get(i).unmarkDone();
     }
 
     public int getNumberOfTasks() {
```
In the `TaskList` class, the method `getNumberOfTasks` is well-named and accurately conveys that it returns a count of tasks. This avoids any potential confusion that could arise from shorter or ambiguous names like `taskCount`. Keeping names clear and unambiguous is key to maintainable code.

publicmain commented on src/main/java/milutrock/MiluTrock.java:
```diff
@@ -4,6 +4,10 @@
+ * A task management program that interacts with users through a
+ * command-line interface.
+ */
 public class MiluTrock {
```
The class name `MiluTrock` seems unique but does not convey any meaningful information about what the class does. A more descriptive name could enhance the readability and maintainability of the code. For example, if this class is responsible for managing tasks, a name like `TaskManager` might be more appropriate.

publicmain commented on src/main/java/milutrock/MiluTrock.java:
```diff
@@ -17,7 +21,7 @@ public MiluTrock(String path) {
+    private void run() {
         ui.printBanner();
 
         storage.loadTasks();
```
The comments added are helpful in understanding the purpose of each method. However, ensure that the comments stay consistent with the method's name and purpose. For example, the comment for the `loadTasks` method is good but could be improved by explicitly stating that it reads tasks from a file.

publicmain commented on src/main/java/milutrock/MiluTrock.java:
```diff
@@ -17,7 +21,7 @@ public MiluTrock(String path) {
     }
 
-    public void run() {
+    private void run() {
```
The method `run` is concise and accurately describes what it does. However, since this method was changed to private, it might be beneficial to consider a more specific name that reflects its scope and usage, such as `executeTasks` or `startProcessing`.

publicmain reviewed (commented):
Overall, the naming conventions used in this code are decent, but there is room for improvement. More descriptive and specific names for classes and methods would significantly enhance code readability and maintainability.

colinhia commented on src/main/java/milutrock/Parser.java:
```diff
@@ -0,0 +1,124 @@
+            words[0].equals("todo")
+            || words[0].equals("deadline")
+            || words[0].equals("event")
+        ) {
```
I like how neatly idented the if-else blocks are here

colinhia commented on src/main/java/milutrock/Storage.java:
```diff
@@ -0,0 +1,65 @@
+import java.io.OutputStream;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
```
Should the import statements be in alphabetical order?

colinhia commented on src/main/java/milutrock/Parser.java:
```diff
@@ -0,0 +1,124 @@
+    }
+
+    /**
+     * Returns the value of the `stdin` variable.
```
Perhaps the comment could be aligned with the one below, where it specifies user input over stdin

colinhia reviewed (commented):
Overall coding style looks good! Good job! Just some minor nitpicks

Commit:
Migrate to GUI

MiloTruck committed:
Merge pull request #8 from MiloTruck/branch-Level-10

Migrate to GUI

Commit:
Modify build.gradle to add mainClassName

mainClassName is set to Launcher class for shadowJar. This is to
accommodate the migration from CLI to GUI.

Commit:
Add assertions to Parser and TaskList

Several implicit assumptions must hold true for parsing and the
tasklist while the program is running. Assert statements have been added to ensure these assumptions hold during testing.

Commit:
Refactor to improve code quality

Code has been refactored to add appropriate spacing and avoid
complicated expressions. This helps to improve readability.

MiloTruck committed:
Merge pull request #9 from MiloTruck/branch-A-Assertions

Add assertions to Parser and TaskList

MiloTruck committed:
Merge pull request #10 from MiloTruck/branch-A-CodeQuality

Refactor to improve code quality

Commit:
Implement multi-delete functionality

Provide a way to delete multiple tasks in one go, by separating their
indexes with a whitespace.

Commit:
Add user documentation

Commit:
Remove ip submodule

Commit:
Add title to GUI window
